pkg/hydrate/controller.go

```diff
@@ -312,24 +316,25 @@ func SourceCommitAndDir(sourceType v1beta1.SourceType, sourceRoot cmpath.Absolut
 		containerName = reconcilermanager.HelmSync
 	}
 
-	// A function that turns an error to a status sourceError.
-	toSourceError := func(err error) status.Error {
-		if err == nil {
-			err = errors.Errorf("unable to sync repo\n%s",
-				git.SyncError(containerName, errFilePath, fmt.Sprintf("%s=%s", metadata.ReconcilerLabel, reconcilerName)))
-		} else {
-			err = errors.Wrapf(err, "unable to sync repo\n%s",
-				git.SyncError(containerName, errFilePath, fmt.Sprintf("%s=%s", metadata.ReconcilerLabel, reconcilerName)))
-		}
-		return status.SourceError.Wrap(err).Build()
+	content, err := os.ReadFile(errFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return "", "", status.TransientError(
+			fmt.Errorf("unable to sync repo\nunable to load %s: %v. Please "+
```

Comment:
do we need the linebreak here? Would a `"` suffice?

Reply:
The `unable to sync repo` doesn't provide any value, so I removed that.